### DISCUSSION TITLE
Index for HTML-Files

### DISCUSSION THREAD

**Discussion by benjaminschlegel** [quarto-dev/quarto-cli#3849] [Feature Requests]
I really like to be able to create an index in Quarto book with \index{word}. At the moment it just works and displays in PDF-Outputs. It would be nice if it would also work for html documents and be added as an extra document at the end of the quarto website.

**Comment by baptiste** [comment]
I've been thinking about this too. My first idea was to switch from `\index{concept}` to a language agnostic syntax such as `[concept]{.index}` and use the [latex-environment](https://github.com/quarto-ext/latex-environment) to do the conversion. I then thought it might be more convenient to have a [Lua filter](https://groups.google.com/g/pandoc-discuss/c/pxW6y3tqyLo/m/ygK6iyLHAQAJ) detect automatically specific words in the manuscript. At that point, I naively concluded that the idea of an index in HTML format was of little use, since one could simply use the website's built-in search function. 

_However_, I've since realised how wrong I was when reading [this blogpost](https://www.arthurperret.fr/blog/2023-02-25-indexer-sa-these.html): a brute-force listing of _all_ appearances of a keyword is not really an index, but more of a _concordance_.  An index necessarily involves some hard work by the author, to identify a helpful structure for the reader. In that sense, it's as useful in HTML as it is in hard copy format.

All this to say: I do not know of a HTML solution to your question, but I think it'd be a great addition to Quarto! (especially if an output-agnostic convention is developed, that works across formats).

As far as a HTML output is concerned, I think something like this would look quite nice: https://codepen.io/gcyrillus/pen/kNjOjw
![Screenshot 2023-02-26 at 16 24 04](https://user-images.githubusercontent.com/18970/221390663-291517c4-ee2b-4145-a2ab-0fe8f328281b.png)

**Reply by benjaminschlegel** [reply]
I don't know how Quarto works in the background, but maybe something like this could work: replace during rendering for HTML all \index{BlaBla} by an HTML-anker and save the word with the file_name and the anker in some sort of list. At the end sort the list and output it as linkes in a new document at the end. Somehow multiple occurence of a word should be grouped.

**Reply by baptiste** [reply]
The span notation `[concept]{.index}` would take care of the first part, and to build the HTML index one could use [this SO answer](https://stackoverflow.com/a/65656823), but I haven't been successful in running the filter in practice (I'm guessing Pandoc has changed a bit since then).
